modules/appeals_api/app/services/appeals_api/evidence_submission_request_validator.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module AppealsApi
+  class EvidenceSubmissionRequestValidator
+    def initialize(nod_id, request_ssn)
+      @nod_id = nod_id
+      @request_ssn = request_ssn
+    end
+
+    def call
+      return [:error, record_not_found_error] unless notice_of_disagreement_exists?
+      return [:error, invalid_review_option_error] unless evidence_accepted?
+      return [:error, invalid_veteran_id_error] unless ssn_match?
+
+      [:ok, {}]
+    end
+
+    private
+
+    def notice_of_disagreement_exists?
+      @notice_of_disagreement ||= AppealsApi::NoticeOfDisagreement.find_by(id: @nod_id)
+    end
+
+    def evidence_accepted?
+      @notice_of_disagreement.accepts_evidence?
+    end
+
+    def ssn_match?
+      # if PII expunged not validating for matching SSNs
+      return true unless @notice_of_disagreement.auth_headers
```

Comment:
how does the presence of auth_headers denote whether PII is expunged or not?

Reply:
`def remove_pii!
      records_to_be_expunged.update(form_data: nil, auth_headers: nil)
 end`

Reply:
Oh for sure, wasn't 100% on how that worked, thanks for the reminder.